code/src/test/kotlin/com/expediagroup/sdk/core/logging/masking/LogMaskerTest.kt

```diff
@@ -0,0 +1,148 @@
+package com.expediagroup.sdk.core.logging.masking
+
+import com.ebay.ejmask.api.MaskingPattern
+import com.expediagroup.sdk.core.logging.common.Constant.OMITTED
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+
+class LogMaskerTest {
+    @AfterEach
+    fun afterEach() {
+        mask.clear()
+    }
+
+    @Nested
+    inner class FieldLogMaskingFeatureTest {
+        @Test
+        fun `adding global fields to the masking pattern builder`() {
+            val globalFields = setOf("globalField1", "globalField2")
+            val patterns = MaskingPatternBuilder().globalFields(*globalFields.toTypedArray()).build()
+
+            assertEquals(1, patterns.size)
+
+            assertTrue(patterns.first().pattern.pattern().contains("globalField1"))
+            assertTrue(patterns.first().pattern.pattern().contains("globalField2"))
+        }
+
+        @Test
+        fun `adding path fields to the masking pattern builder`() {
+            val pathFields = setOf(listOf("first", "second"), listOf("first1", "second1", "third1", "as"))
+            val patterns = MaskingPatternBuilder().pathFields(*pathFields.toTypedArray()).build()
+
+            assertEquals(pathFields.size, patterns.size)
+        }
+
+        @Test
+        fun `adding global and path fields to the masking pattern builder`() {
+            val globalFields = setOf("globalField1", "globalField2")
+            val pathFields = setOf(listOf("first", "second"), listOf("first1", "second1", "third1", "as"))
+
+            val maskingPatterns = MaskingPatternBuilder()
+                .globalFields(*globalFields.toTypedArray())
+                .pathFields(*pathFields.toTypedArray())
+                .build()
+
+            assertEquals(pathFields.size + 1, maskingPatterns.size)
```

Comment:
do we need to add similar assert for `globalFields`? maybe assert their values also?

Reply:
The `+1` asserts that. This is correct because regardless of how many global fields you add, they will be consumed in a single generated regex. I will add a docstring to make it clear.